sklearn/datasets/_samples_generator.py

```diff
@@ -1568,19 +1550,8 @@ def make_sparse_coded_signal(
     # encode signal
     Y = np.dot(D, X)
 
-    # TODO(1.5) remove data_transposed
-    # raise warning if data_transposed is not passed explicitly
-    if data_transposed != "deprecated":
-        warnings.warn(
-            "data_transposed was deprecated in version 1.3 and will be removed in 1.5.",
-            FutureWarning,
-        )
-    else:
-        data_transposed = False
-
-    # transpose if needed
-    if not data_transposed:
-        Y, D, X = Y.T, D.T, X.T
+    # Transpose to have shapes consistent with the rest of the API
+    Y, D, X = Y.T, D.T, X.T
```

Comment:
Full-disclosure: I have not tried to undersand the back-story behind this, but would it make sense to generate `D` and `X` already transposed and do `Y = np.dot(X, D)` so that it is slightly easier to match what the docstring says?

Reply:
I would have liked to but we decided not to because it would change the results and we usually try to avoid that is possible.

Reply:
OK fair enough